src/main/java/com/goit/hotelonlinebooking/dao/AbstractUserDAO.java

```diff
@@ -0,0 +1,23 @@
+package com.goit.hotelonlinebooking.dao;
+
+import com.goit.hotelonlinebooking.entity.User;
+
+import java.util.List;
+
+/**
+ * Created by user on 28.10.2016.
+ */
+public abstract class AbstractUserDAO extends AbstractDAO {
```

Comment:
UserDAO should not be abstract. You should implement all methods. Add a collection with users that will be used as a storage.